### ISSUE TITLE
Support transformation of configuration values

### ISSUE THREAD

**Issue by gastaldi** [eclipse/microprofile-config#451]
A common use-case is when encrypted values are used in a properties file (sensitive values, like the database password). It would be nice if the spec supported an SPI for transformation of configuration properties.

**Comment by gastaldi** [comment]
One way to implement that is to introduce a special type (`SensitiveValue`) and a converter for that type

**Comment by Emily-Jiang** [comment]
Do you mean you want to decrypt the value? I think you will have to encrypt the sensitive info and then you can use special utility from runtime to decrypt the value.

**Comment by gastaldi** [comment]
@Emily-Jiang that's correct. But encrypting (writing) the value is out of scope of the Config profile

**Comment by dmlloyd** [comment]
Presumably a custom converter would be able to achieve this.

**Comment by gastaldi** [comment]
@dmlloyd right, but using a custom Converter (String -> String if I understand correctly) is not enough when another type is used as the target. I presume only one Converter is invoked per type?

**Comment by dmlloyd** [comment]
Only one converter, correct, however there's nothing stopping a converter from delegating to another converter.

In #445 I've proposed that a converter should be able to be provided per call to `getValue`, so users can perform custom non-global conversions; I've also proposed a method to get the registered global converter for a given type.  In Quarkus we have a `@ConvertWith` annotation which allows a custom converter class to be specified.

For the password case, if the desired cleartext format password is `String` (as opposed to `char[]` or something like Elytron's `ClearPassword` or similar) then yes, you'd implement a `Converter<String>` which either delegates to the default converter (through the method I mentioned before) or simply does the conversion directly (given that the system `Converter<String>` is fairly trivial).
